packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1200,26 +1200,39 @@ function pushEffect(tag, create, destroy, deps) {
 function mountRef<T>(initialValue: T): {|current: T|} {
   const hook = mountWorkInProgressHook();
   if (__DEV__) {
-    const fiber = currentlyRenderingFiber;
+    // Support lazy initialization pattern shown in docs.
+    let hasBeenInitialized = initialValue !== null;
```

Comment:
What about `undefined`? Many people who don't use type systems just do `useRef()`.

Reply:
Thought about that. Was concerned about this case though:
```js
const ref = useRef(someVarWithPossiblyUndefinedValueThatMightChange);
```
I think it would be okay if we treated `initialValue === null || arguments.length === 0` though.

Reply:
I guess I'm curious why `someVarWithPossiblyUndefinedValueThatMightChange` wouldn't be a problem for `null`.

Reply:
It could be, but I couldn't think of a way to differentiate/detect the lazy init pattern that wouldn't false positive :)